src/Gateway.php

```diff
@@ -116,6 +124,56 @@ public function payment_fields() {
 		);
 	}
 
+	/**
+	 * Add additional billing fields to the checkout form.
+	 *
+	 * @param array $fields The billing fields.
+	 * @return array
+	 */
+	public function additional_billing_fields( $fields ) {
+
+		// NOTE: If you'll be changing the property name, make sure to update it in the JavaScript code too (file: assets/js/ledyer-payments.js).
+		$fields['billing_customer_reference1'] = array(
+			'label'       => __( 'Reference 1', 'ledyer-payments-for-woocommerce' ),
+			'placeholder' => __( 'For example, reference code or cost center', 'ledyer-payments-for-woocommerce' ),
+			'required'    => false,
+			'class'       => array( 'form-row-wide' ),
+			'clear'       => true,
+			'priority'    => 35,
+		);
+
+		$fields['billing_customer_reference2'] = array(
+			'label'       => __( 'Reference 2', 'ledyer-payments-for-woocommerce' ),
+			'placeholder' => __( 'For example, purchase order number', 'ledyer-payments-for-woocommerce' ),
+			'required'    => false,
+			'class'       => array( 'form-row-wide' ),
+			'clear'       => true,
+			'priority'    => 35,
+		);
```

Comment:
You are adding these fields to the billing address fields instead of bellow the company number as the task says. Is there a reason for that?

Reply:
It is appearing below the company number field though. Is this not happening for you?

![image](https://github.com/user-attachments/assets/350021b0-a999-46cd-a7b4-57f0a554c790)

Reply:
No i am getting it showing like this:
![image](https://github.com/user-attachments/assets/271a965d-9b50-4efd-a189-5ebcf42d4559)

Reply:
Oh, I should have been more clear. What you're seeing is the intended behavior when the setting "Company number field" is set to the default `Payment form` value. You need to set it to `Billing form` instead.

Reply:
After discussing with Ledyer, we've decided to keep everything within the billing form field, including the company number field. I'll create a new PR for that, so I think we can merge this one for now.